mono/utils/mono-context.h

```diff
@@ -129,18 +154,42 @@ typedef struct {
 #else
 #define MONO_CONTEXT_GET_CURRENT(ctx) \
 	__asm__ __volatile__(   \
-	"movl $0x0, 0x00(%0)\n" \
-	"mov %%ebx, 0x04(%0)\n" \
-	"mov %%ecx, 0x08(%0)\n" \
-	"mov %%edx, 0x0c(%0)\n" \
-	"mov %%ebp, 0x10(%0)\n" \
-	"mov %%esp, 0x14(%0)\n" \
-	"mov %%esi, 0x18(%0)\n" \
-	"mov %%edi, 0x1c(%0)\n" \
+	"movl $0x0, %c[eax](%0)\n" \
```

Comment:
The `__asm__` features are not very well documented. I hadn’t seen this before, either.